cmd/commands/function.go

```diff
@@ -46,51 +46,44 @@ func FunctionCreate(fcTool *core.Client) *cobra.Command {
-		"command":         "https://github.com/projectriff/command-function-invoker/raw/v0.0.7/command-invoker.yaml",
-		"command-invoker": "https://github.com/projectriff/command-function-invoker/raw/v0.0.7/command-invoker.yaml",
-		"node":            "https://github.com/projectriff/node-function-invoker/raw/v0.0.8/node-invoker.yaml",
-		"node-invoker":    "https://github.com/projectriff/node-function-invoker/raw/v0.0.8/node-invoker.yaml",
+		"jar":     "https://github.com/projectriff/java-function-invoker/raw/v0.1.1/java-invoker.yaml",
+		"command": "https://github.com/projectriff/command-function-invoker/raw/v0.0.7/command-invoker.yaml",
+		"node":    "https://github.com/projectriff/node-function-invoker/raw/v0.0.8/node-invoker.yaml",
 	}
 
 	command := &cobra.Command{
 		Use:   "create",
 		Short: "Create a new function resource",
 		Long: "Create a new function resource from the content of the provided Git repo/revision or local source.\n" +
-			"\nThe RUNTIME arg defines the language runtime that is added to the function code in the build step. The resulting image is then used to create a Knative Service (`service.serving.knative.dev`) instance of the name specified for the function. The following runtimes are available:\n\n" +
-			// TODO make runtime help data-driven
-			"- 'java': uses riff's java-function-invoker (aliased as java-invoker)\n" +
-			"- 'node': uses riff's node-function-invoker (aliased as node-invoker)\n" +
-			"- 'command': uses riff's command-function-invoker (aliased as command-invoker)\n" +
-			"- 'java-buildpack': uses the riff Buildpack \n" +
-			"- 'detect': uses the riff Buildpack's detection (currently limited to Java functions) \n" +
-			"\nClassic riff Invoker runtimes are available in addition to experimental Buildpack runtimes.\n" +
-			"\nBuildpack based runtimes support building from local source in addition to within the cluster. Locally built images prefixed with 'dev.local/' are saved to the local Docker daemon while all other images are pushed to the registry specified in the image name.\n" +
+			"\nThe INVOKER arg defines the language runtime that is added to the function code in the build step. The resulting image is then used to create a Knative Service (`service.serving.knative.dev`) instance of the name specified for the function. The following invokers are available:\n\n" +
+			"- 'java': uses the riff Buildpack to build Maven or Gradle projects from source\n" +
+			"- 'jar': uses riff's java-function-invoker build for a prebuilt JAR file\n" +
+			"- 'node': uses riff's node-function-invoker build\n" +
+			"- 'command': uses riff's command-function-invoker build\n" +
+			"\nBuildpack based builds support building from local source in addition to within the cluster. Locally built images prefixed with 'dev.local/' are saved to the local Docker daemon while all other images are pushed to the registry specified in the image name.\n" +
```

Comment:
should this say "are *also* pushed to the registry? Usage of something other than `dev.local` still requires a local daemon

Reply:
builds are either published to a remote registry or the local docker daemon, not both

Reply:
Builds always happen in the local daemon (and hence are available there) and *then* maybe are pushed to a (remote) registry. The code clearly uses a boolean flag named `publish`. My point is that when you use `dev.local`, there is no extra push

Reply:
I don't think non `dev.local` builds use or require the daemon, they interact directly with the registry

Reply:
Try it :)

Reply:
I guess it does require the daemon ;)

Reply:
s/in addition to/or/

I would change the "Locally built..." sentence to:

> Images will be pushed to the registry specified in the image name, unless prefixed with 'dev.local/' in which case the image will only be available within the local Docker daemon."

Reply:
the non `dev.local` images are pushed to the registry and not to the local daemon, at least I can't find them